tests/org.palladiosimulator.editors.sirius.tests/src/org/palladiosimulator/editors/sirius/tests/AnchorProviderTest.java

```diff
@@ -0,0 +1,695 @@
+            .toArray()[0];
+
+        assertNotNull(this.diagram);
+
+        // Set up Default Cursor
+        this.cursor = new LocationRequest();
+        this.cursor.setLocation(new Point(690, 237));
+
+        // run testing editor
+        this.editorPart = DialectUIManager.INSTANCE.openEditor(this.session, this.diagram, new NullProgressMonitor());
+        TestsUtil.synchronizationWithUIThread();
+    }
+
+    /**
+     * Closes the editor and tears down all used resources at the end.
+     */
+    @Override
+    protected void tearDown() throws Exception {
+        DialectUIManager.INSTANCE.closeEditor(this.editorPart, false);
+        TestsUtil.synchronizationWithUIThread();
+        this.diagram = null;
+        this.editorPart = null;
+        super.tearDown();
+    }
+
+    /**
+     * Data Handler for Points and Diagram Elements of a Connection. <br>
+     * <br>
+     * 1) Use the constructor to create a connection edge. <br>
+     * 2) Use the assertion methods for checking values of the Points. <br>
+     * <br>
+     * 3) Use deleteEdge() for cleaning up the testmodel diagram for the next test.
+     *
+     * @author Jonas Lehmann
+     */
+    private class ConnectionInitializer {
+
+        private Point srcAnchorPoint, srcEdgeEndPoint, targetAnchorPoint, targetEdgeEndPoint;
+
+        private final String nameOfEdgeCreationTool;
+
+        private DDiagramElement srcNode, targetNode;
+        private NodeEditPart srcNodeEP, targetNodeEP;
+        private Rectangle srcNodeRect, targetNodeRect;
+
+        private DEdge dEdge;
+        private Connection connection;
+
+        /**
+         * Creates a new {@link ConnectionInitializer}.
+         *
+         * @param srcName
+         *            the name of the starting node in the assembly diagram
+         * @param targetName
+         *            the name of the target node in the assembly diagram
+         * @param nameOfEdgeCreationTool
+         *            the name of the connector needed to connect the two nodes
+         */
+        public ConnectionInitializer(final String srcName, final String targetName,
+                final String nameOfEdgeCreationTool) {
+            this(srcName, targetName, nameOfEdgeCreationTool, true);
+        }
+
+        /**
+         * Alternative Constructor to run/skip comparison with edge end points.
+         *
+         * @param srcName
+         *            the name of the starting node in the assembly diagram
+         * @param targetName
+         *            the name of the target node in the assembly diagram
+         * @param nameOfEdgeCreationTool
+         *            the name of the connector needed to connect the two nodes
+         * @param compareWithEdgeEndPoints
+         *            if comparison is needed set true
+         */
+        public ConnectionInitializer(final String srcName, final String targetName, final String nameOfEdgeCreationTool,
```

Comment:
Is absolutely fine and understand it also why the `ConnectionInitializer` was implemented as inner class. However, for future tests and or extensions this class could be implemented as a separate compilation unit.